packages/app-desktop/gui/ToolbarBase.tsx

```diff
@@ -1,88 +1,129 @@
 			paddingRight: theme.mainPadding,
-		}, this.props.style);
-
-		const groupStyle: any = {
-			display: 'flex',
-			flexDirection: 'row',
-			boxSizing: 'border-box',
-		};
-
-		const leftItemComps: any[] = [];
-		const centerItemComps: any[] = [];
-		const rightItemComps: any[] = [];
-
-		if (this.props.items) {
-			for (let i = 0; i < this.props.items.length; i++) {
-				const o = this.props.items[i];
-				let key = o.iconName ? o.iconName : '';
-				key += o.title ? o.title : '';
-				key += o.name ? o.name : '';
-				const itemType = !('type' in o) ? 'button' : o.type;
-
-				if (!key) key = `${o.type}_${i}`;
-
-				const props = Object.assign(
-					{
-						key: key,
-						themeId: this.props.themeId,
-					},
-					o
-				);
-
-				if (o.name === 'toggleEditors') {
-					rightItemComps.push(<ToggleEditorsButton
+			overflow: 'hidden',
+		},
+		Props.style
+	);
+	const groupStyle: any = {
+		display: 'flex',
+		flexDirection: 'row',
+		boxSizing: 'border-box',
+	};
+	const leftItemComps: any[] = [];
+	const centerItemComps: any[] = [];
+	const rightItemComps: any[] = [];
+	if (Props.items) {
+		for (let i = 0; i < Props.items.length; i++) {
+			const o = Props.items[i];
+			let key = o.iconName ? o.iconName : '';
+			key += o.title ? o.title : '';
+			key += o.name ? o.name : '';
+			const itemType = !('type' in o) ? 'button' : o.type;
+			if (!key) key = `${o.type}_${i}`;
+			const props = Object.assign(
+				{
+					key: key,
+					themeId: Props.themeId,
+				},
+				o
+			);
+			if (o.name === 'toggleEditors') {
+				rightItemComps.push(
+					<ToggleEditorsButton
 						key={o.name}
 						value={Value.Markdown}
-						themeId={this.props.themeId}
+						themeId={Props.themeId}
 						toolbarButtonInfo={o}
-					/>);
-				} else if (itemType === 'button') {
-					const target = ['historyForward', 'historyBackward', 'toggleExternalEditing'].includes(o.name) ? leftItemComps : centerItemComps;
-					target.push(<ToolbarButton {...props} />);
-				} else if (itemType === 'separator') {
-					centerItemComps.push(<ToolbarSpace {...props} />);
-				}
+					/>
+				);
+			} else if (itemType === 'button') {
+				const target = [
+					'historyForward',
+					'historyBackward',
+					'toggleExternalEditing',
+				].includes(o.name)
+					? leftItemComps
+					: centerItemComps;
+				target.push(<ToolbarButton {...props} />);
+			} else if (itemType === 'separator') {
+				centerItemComps.push(<ToolbarSpace {...props} />);
 			}
 		}
-
-		return (
-			<div className="editor-toolbar" style={style}>
-				<div style={groupStyle}>
-					{leftItemComps}
-				</div>
-				<div style={groupStyle}>
-					{centerItemComps}
-				</div>
-				<div style={Object.assign({}, groupStyle, { flex: 1, justifyContent: 'flex-end' })}>
-					{rightItemComps}
-				</div>
-			</div>
-		);
 	}
+	return (
+		<div className="editor-toolbar" ref={parentRef} style={style}>
+			{React.createElement(horizScroll ? HorizontalScroll : 'div', {
```

Comment:
We don't use React.createElement, please use JSX

Reply:
Actually I used React.createElement to prevent too many changes (otherwise I would need to separate children into another variable).
So would it be better to use
```typescript
{horizScroll ? (
	<HorizontalScroll reverseScroll style={{ display: 'flex' }}>
		{children}
	</HorizontalScroll>
) : (
	children
)}
```

Reply:
changed to JSX in my commit